storage/store.go

```diff
@@ -1382,14 +1387,25 @@ func (s *Store) resolveWriteIntentError(ctx context.Context, wiErr *proto.WriteI
 			},
 		}
 		resolveReply := &proto.InternalResolveIntentResponse{}
-		// Add resolve command with wait=false to add to Raft but not wait for completion.
-		waitForResolve := wait && i == len(wiErr.Intents)-1
-		if resolveErr := rng.AddCmd(ctx, proto.Call{Args: resolveArgs, Reply: resolveReply}, waitForResolve); resolveErr != nil {
-			if log.V(1) {
-				log.Warningc(ctx, "resolve for key %s failed: %s", intent.Key, resolveErr)
+
+		go func(errCh chan<- error, key proto.Key) {
+			if resolveErr := rng.AddCmd(ctx, proto.Call{Args: resolveArgs, Reply: resolveReply}, true); resolveErr != nil {
+				if log.V(1) {
+					log.Warningc(ctx, "resolve for key %s failed: %s", key, resolveErr)
+				}
+				errCh <- resolveErr
 			}
-			return resolveErr
-		}
+			wg.Done()
+		}(errChan, intent.Key)
+	}
+
+	go func() {
+		wg.Wait()
```

Comment:
sure, made this change. I haven't the faintest idea how to test any of this though, any thoughts?